evm/src/cpu/kernel/interpreter.rs

```diff
@@ -659,6 +685,103 @@ impl<'a> Interpreter<'a> {
         }
     }
 
+    fn run_codesize(&mut self) {
+        self.push(
+            self.generation_state.memory.contexts[self.context].segments
+                [Segment::ContextMetadata as usize]
+                .get(ContextMetadata::CodeSize as usize),
+        )
+    }
+
+    fn run_codecopy(&mut self) {
+        let dest_offset = self.pop().as_usize();
+        let offset = self.pop().as_usize();
+        let size = self.pop().as_usize();
+        for i in 0..size {
+            let code_byte =
+                self.generation_state
+                    .memory
+                    .mload_general(self.context, Segment::Code, offset + i);
+            self.generation_state.memory.mstore_general(
+                self.context,
+                Segment::MainMemory,
+                dest_offset + i,
+                code_byte,
+            );
+        }
+    }
+
+    fn run_gasprice(&mut self) {
+        self.push(self.get_txn_field(NormalizedTxnField::ComputedFeePerGas))
+    }
+
+    fn run_extcodesize(&mut self) {
+        let offset = self.pop().as_usize();
+        // TrieData stores for each address the tuple (nonce, balance, storage_root, code_hash).
+        let code_hash = self.get_trie_data()[offset..][3];
+        self.push(
+            self.generation_state
+                .inputs
+                .contract_code
+                .get(&H256::from_uint(&code_hash))
+                .unwrap_or(&vec![])
+                .len()
+                .into(),
+        );
+    }
+
+    fn run_extcodecopy(&mut self) {
+        let dest_offset = self.pop().as_usize();
+        let offset = self.pop().as_usize();
+        let size = self.pop().as_usize();
+        for i in 0..size {
+            let extcode_byte = self.generation_state.memory.mload_general(
+                0,
+                Segment::KernelAccountCode,
```

Comment:
I think this should be `self.context` and `Segment::Code`

Reply:
I guess I got confused by [this line](https://github.com/mir-protocol/plonky2/blob/b721236eb3a9e069d79de3f37dd384c2dafd08e9/evm/src/cpu/kernel/asm/account_code.asm#L130) in the kernel

Reply:
Thanks for the explanation William, and yes indeed I had missed the first pop. Following your recommendation I'll leave it unimplemented.